homeassistant/components/tplink_omada/config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Config flow for TP-Link Omada integration."""
+from __future__ import annotations
+
+import logging
+from types import MappingProxyType
+from typing import Any
+
+from tplink_omada_client.exceptions import (
+    ConnectionFailed,
+    LoginFailed,
+    OmadaClientException,
+    UnsupportedControllerVersion,
+)
+from tplink_omada_client.omadaclient import OmadaClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, CONF_VERIFY_SSL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_VERIFY_SSL, default=True): bool,
+        vol.Required("site", default="Default"): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class OmadaHub:
+    """Omada Controller hub."""
+
+    def __init__(self, hass: HomeAssistant, data: MappingProxyType[str, Any]) -> None:
+        """Initialize."""
+        self.hass = hass
+        self.host = data[CONF_HOST]
+        self.site = data["site"]
+        self.verify_ssl = bool(data[CONF_VERIFY_SSL])
+        self.username = data[CONF_USERNAME]
+        self.password = data[CONF_PASSWORD]
+        self.client = None
+
+    def get_client(self) -> OmadaClient:
+        """Get the client api for the hub."""
+        if not self.client:
+            websession = async_get_clientsession(self.hass, verify_ssl=self.verify_ssl)
+            self.client = OmadaClient(
+                self.host,
+                self.username,
+                self.password,
+                websession=websession,
+                site=self.site,
+            )
+        return self.client
+
+    async def authenticate(self) -> bool:
+        """Test if we can authenticate with the host."""
+
+        client = self.get_client()
+        await client.login()
+
+        return True
+
+    async def get_controller_name(self) -> str:
+        """Get the display name of the controller."""
+        client = self.get_client()
+        name = await client.get_controller_name()
+        return str(name)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    hub = OmadaHub(hass, MappingProxyType(data))
+
+    await hub.authenticate()
+    name = await hub.get_controller_name()
+
+    # Return info that you want to store in the config entry.
+    return {"title": f"TP-Link Omada {name} ({data['site']})"}
```

Comment:
The title can just be `name`. It will already be shown on the integration card next to the TP-Link logo and integration name.

Reply:
Makes sense.